internal/weapons/catalyst/hakushin/hakushin.go

```diff
@@ -0,0 +1,47 @@
+package hakushin
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+func init() {
+	core.RegisterWeaponFunc("hakushin ring", weapon)
+	core.RegisterWeaponFunc("hakushin", weapon)
+}
+
+func weapon(char core.Character, c *core.Core, r int, param map[string]int) string {
+
+	expiry := 0
+	e := 10 + float64(r)*2.5
```

Comment:
This seems wrong on two levels, first you don't divide this by 100 anywhere so this is going to be a 1000% buff...

Second the R1 buff is 10% not 12.5%